browser/ui/sidebar/sidebar_service_factory.cc

```diff
@@ -31,15 +32,42 @@ SidebarServiceFactory::SidebarServiceFactory()
 
 SidebarServiceFactory::~SidebarServiceFactory() = default;
 
+std::vector<SidebarItem::BuiltInItemType>
+SidebarServiceFactory::GetBuiltInItemTypesForProfile(Profile* profile) const {
+  std::vector<SidebarItem::BuiltInItemType> types;
+  for (const auto& type : kDefaultBuiltInItemTypes) {
+    if (profile->IsGuestSession()) {
+      if (!IsDisabledItemForGuest(type)) {
+        types.push_back(type);
+      }
+      continue;
+    }
+
+    // Private(or tor) profile.
```

Comment:
This is not correct. `IsIncognitoProfile` is only the private window profile and does not include `Tor`. To get both you need to use `profile->IsOffTheRecord`

Reply:
There should be tests for this